js/app.js

```diff
@@ -13,34 +13,55 @@ var fig_cfg = {
     },
     legend_cfg = {
       translate_x: 300,
-      translate_y: 40,
+      translate_y: 35,
       circle_radius: 10,
       num_bins: 17
+    },
+    z_indices = {
+      underlayPlaceholder: -100,
+      mainCanvas: 10,
+      mapExtras: 20,
+      overlayStates: 30,
+      legend: 40,
+      overlayLegend: 50,
+      siteHighlighter: 100
     };
 
 // use existing svg element and add some attributes
 d3.select("#mainFig").select("svg")
-    .attr("id", "plotarea")
-    .style("z-index", -10)
+    .attr("id", "mapSvg")
     .style("position", "absolute");
 
+// Make a second svg to put the legend and siteHighlighter into
+// This will allow the map to be below the canvas points,
+// But the siteHighlighter and legend to be above the canvas layer
+var svgwidth = d3.select('#mapSvg').attr("width"),
+    svgheight = d3.select('#mapSvg').attr("height");
+d3.select('#mainFig').append('svg')
+  .attr("id", "mapExtras")
+  .attr("width", svgwidth)
+  .attr("height", svgheight)
+  .style("position", "absolute")
+  .style("z-index", z_indices.mapExtras);
+
 // create canvas layer
 var canvas = d3.select("#mainFig").append('canvas')
       .attr('id', "mainCanvas")
       .attr('width', fig_cfg.width)
-      .attr('height', fig_cfg.width)
-      .style("position", "absolute");
+      .attr('height', fig_cfg.height)
+      .style("position", "absolute")
+      .style("z-index", z_indices.mainCanvas);
 
 var canvas_context = canvas.node().getContext('2d'),
     scale_colors_fxns = create_color_scale_function(legend_cfg),
     dv_stats_data = load_dv_data();
 
 // add different figure features
-add_color_legend(scale_colors_fxns, legend_cfg);
-add_circle_selector();
-clone_states(fig_cfg);
-clone_legend(fig_cfg, legend_cfg);
-add_placeholder(fig_cfg);
+add_color_legend(scale_colors_fxns, legend_cfg, z_indices);
+add_circle_selector(z_indices);
+clone_states(fig_cfg, z_indices);
+clone_legend(fig_cfg, legend_cfg, z_indices);
+add_placeholder(fig_cfg, z_indices);
```

Comment:
I think it could be argued either way, but for the record, I would have used z_indices differently - I would pass z_indices.siteHighlighter to add_circle_selector, pass z_indices.underlayPlaceholder to add_placeholder, etc. rather than needing to know the name of the relevant z index within each of those functions. I think this would help consolidate the need to understand the z_indices structure within this one app.js file rather than causing that info to be distributed across all the modules as well, and that would in turn make the code just a touch more modifiable. But it'd be a minor improvement at best, and I'm not sure if there's a clear-cut best practice in anybody's mind, so just take this as food for thought for future code.

Reply:
Hmm I think that is a good idea - I'm going to go ahead and switch it now since it's not too difficult!